### ISSUE TITLE
Aggregate expressions in qfield

### ISSUE THREAD

**Issue by RatInTheField** [opengisch/QField#828]
[Short description of problem here]
I use aggregate expressions to default populate fields in QGIS successfully. When migrated to qfield, they do not perform as expected.

in QGIS, _to_int(maximum( "qid",filter:="CreateBy"=@Operator_name ))+1_ produces a numeric value, but in Qfield it produces a NULL value.

The expression is set as the Default Value of a column.

### Reproduction Steps:

1. [First Step]
Enter new record
2. [Second Step]
Error visible in data entry form
3. [Other Steps...]

### Expected behavior:

[Describe expected behavior here]
In QGIS, default in a numeric value based on the maximum value of a column in a layer filtered by another column.
In Qfield, default in a numeric value based on the maximum value of a column in a layer filtered by another column.

### Observed behavior:
In QGIS, the numeric value is loaded correctly and has the correct value.
In Qfield, the numeric value defaults to NULL

[Describe observed behavior here]

### Screenshots and GIFs

![Screenshots and GIFs which follow reproduction steps to demonstrate the problem](url)

[Please also attach additional files if a specific project/dataset is useful to investigate the problem.]

**QField version:** [Enter QField version here including build number found in the about screen after the release name example 1.2.0 (10200993)]
1.3.4 Ben Nevis (10304995)

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: [Yes/No]
Not known
* Problem can be reliably reproduced, doesn't happen randomly: [Yes/No]
Yes
* Problem happens with all files and projects, not only some files or projects: [Yes/No]
Yes

**Comment by m-kuhn** [comment]
Hi @RatInTheField can you check if this is the same as reported in https://github.com/opengisch/QField/issues/734 ? Thanks

**Comment by RatInTheField** [comment]
Thanks for your response.

My problem is similar, but not identical, I think.

I will check some details then feedback.

**Comment by RatInTheField** [comment]
I have reviewed my issue. I can now consistently create the field values that I need, but cannot recreate the original issue despite all my efforts to replicate it!??

I am sure that this is not #734 as no error messages have been produced.

During the review, I noticed that when referencing a global variable, QGIS is not case sensitive, but Qfield is.
The expression concat(@project_identifier,'_',case when @Operator_name ='Base' then 1
when @operator_name = 'GPS1' then 2
when @operator_name = 'GPS2' then 3
when @operator_name = 'GPS3' then 4
when @operator_name = 'TA' then 5
when @operator_name = 'RatHuawei' then 6
else 0
end
,'_',lpad( maximum( "qid",filter:="CreateBy"=@Operator_name )+1,4,0)) produces 'AC_1_0816' as expected, but in Qfield produces 'AC_0_0804' where I would expect 'AC_6_0804' (AC is the project identifier, 1,2,3,4,5 or 6 refer to the data collection devices running Qfield and 0804 is the maximum value of the qid field filtered by data collection device and incremented by 1. The object is to create a value that remains unique when data from the devices is merged, then returned to the collection devices so that each device can display data from all devices, but issues ids based on their own sequence.

[AngramCommon_FieldDataT4.zip](https://github.com/opengisch/QField/files/4207470/AngramCommon_FieldDataT4.zip)
I have attached a zip of my project. The relevant layer is 'Waypoint', the fields affected are 'qid' and 'name'.
